src/chainstate/stacks/boot/pox-3.clar

```diff
@@ -653,9 +653,11 @@
       (asserts! (check-caller-allowed)
                 (err ERR_STACKING_PERMISSION_DENIED))
 
-      ;; tx-sender principal must not be stacking
-      (asserts! (is-none (get-stacker-info tx-sender))
-        (err ERR_STACKING_ALREADY_STACKED))
+      ;; delegate-stx no longer requires the delegator to not currently
+      ;; be stacking.
+      ;; delegate-stack-* functions assert that
+      ;; 1. users can't swim in two pools at the same time.
+      ;; 2. users can't switch pools without cool down cycle.
```

Comment:
Can we also add that a directly stacked user cannot join a pool without a cool down cycle and vice versa? Then we will also need to add the missing test cases around these situations.